MPSGameLift/Code/Source/MPSGameLiftModuleInterface.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Module/Module.h>
+#include <Unified/MPSGameLiftSystemComponent.h>
+#if AZ_TRAIT_CLIENT
+    #include <MPSGameLiftClientSystemComponent.h>
+#endif
+
+#if AZ_TRAIT_SERVER && !AZ_TRAIT_CLIENT
+    #include <MPSGameLiftServerSystemComponent.h>
+    #include <AzCore/Console/IConsole.h>
+#endif
+
+namespace MPSGameLift
+{
+    class MPSGameLiftModuleInterface
+        : public AZ::Module
+    {
+    public:
+        AZ_RTTI(MPSGameLiftModuleInterface, "{2CB13C68-FA4D-4B94-8CFB-6AC71B8630B6}", AZ::Module);
+        AZ_CLASS_ALLOCATOR(MPSGameLiftModuleInterface, AZ::SystemAllocator, 0);
+
+        MPSGameLiftModuleInterface()
+        {
+            m_descriptors.insert(m_descriptors.end(), {
+                MPSGameLiftSystemComponent::CreateDescriptor(),
+                #if AZ_TRAIT_CLIENT
+                    MPSGameLiftClientSystemComponent::CreateDescriptor(),
+                #endif
+                #if AZ_TRAIT_SERVER && !AZ_TRAIT_CLIENT
+                    MPSGameLiftServerSystemComponent::CreateDescriptor(),
+                #endif
+                });
+        }
+
+        /**
+         * Add required SystemComponents to the SystemEntity.
+         */
+        AZ::ComponentTypeList GetRequiredSystemComponents() const override
+        {
+            AZ::ComponentTypeList requiredSystemComponents{
+                azrtti_typeid<MPSGameLiftSystemComponent>()
+            };
+
+            #if AZ_TRAIT_CLIENT
+                requiredSystemComponents.push_back(azrtti_typeid<MPSGameLiftClientSystemComponent>());
+            #endif
+
+            // Only activate the MultiplayerSample GameLift server system component if this a dedicated server running on GameLift.
+            #if (AZ_TRAIT_SERVER && !AZ_TRAIT_CLIENT)
```

Comment:
Mike [had a suggestion](https://github.com/aws-lumberyard-dev/o3de-multiplayersample/pull/209#discussion_r1144910079) earlier to alias this as a single define with a comment explanation, would that make sense to do earlier in this file?

Reply:
Looks good. Fixed